projects/core/src/site-context/store/effects/languages.effect.ts

```diff
@@ -41,12 +52,26 @@ export class LanguagesEffects {
         this.winRef.sessionStorage.setItem('language', action.payload);
       }
     }),
-    map(() => new SiteContextActions.LanguageChange())
+    withLatestFrom(
+      this.state.select(getActiveLanguage).pipe(
+        bufferCount(2, 1),
+
+        // avoid dispatching `change` action when we're just setting the initial value:
+        filter(([previous]) => !!previous),
+        // avoid dispatching `change` action when values are the same
+        filter(([previous, current]) => previous !== current)
+      )
+    ),
+    map(
+      ([_, [previous, current]]) =>
+        new SiteContextActions.LanguageChange({ previous, current })
```

Comment:
The same case as with "currency change", i.e. why not derive it from the state directly?